tests/Doctrine/Tests/Common/Persistence/Mapping/SymfonyFileLocatorTest.php

```diff
@@ -62,6 +62,90 @@ public function testGetAllClassNames()
         $this->assertEquals(array("Foo\\stdClass"), $locator->getAllClassNames("global"));
     }
 
+
+    public function customNamespaceSeparatorProvider()
+    {
+        return array(
+            'directory separator' => array(DIRECTORY_SEPARATOR, "/_custom_ns/dir"),
+            'underscore separator' => array('_', "/_custom_ns/underscore"),
```

Comment:
TBH, I don't see a problem with supporting `_` here. Even with classes using `_` as pseudo-namespace-separator, the chances of collisions are very low and impossible with default values